OSx compilation with cmake

Hi @marcelm,

I was going to set up a local mini benchmark on my Macbook OSx (Catalina) for `strobealign-0.7.1` vs `strobealign-master` vs `strobealign-WFA` (to be implemented).

I ran into the following installation error on a OSx. Any ideas?

```
(base) math172l2:StrobeAlign ksahlin$ g++ --version
Configured with: --prefix=/Library/Developer/CommandLineTools/usr --with-gxx-include-dir=/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/4.2.1
Apple clang version 11.0.3 (clang-1103.0.32.62)
Target: x86_64-apple-darwin19.6.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin

```

```
(base) math172l2:source ksahlin$ git clone https://github.com/ksahlin/StrobeAlign
Cloning into 'StrobeAlign'...
remote: Enumerating objects: 2290, done.
remote: Counting objects: 100% (358/358), done.
remote: Compressing objects: 100% (166/166), done.
remote: Total 2290 (delta 262), reused 238 (delta 192), pack-reused 1932
Receiving objects: 100% (2290/2290), 1.37 MiB | 1.32 MiB/s, done.
Resolving deltas: 100% (1445/1445), done.
(base) math172l2:source ksahlin$ cd StrobeAlign
(base) math172l2:StrobeAlign ksahlin$ cmake -B build
-- The C compiler identification is AppleClang 11.0.3.11030032
-- The CXX compiler identification is AppleClang 11.0.3.11030032
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found ZLIB: /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/lib/libz.tbd (found version "1.2.11") 
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Found OpenMP_C: -Xclang -fopenmp (found version "3.1") 
-- Found OpenMP_CXX: -Xclang -fopenmp (found version "3.1") 
-- Found OpenMP: TRUE (found version "3.1")  
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/ksahlin/prefix/source/StrobeAlign/build
(base) math172l2:StrobeAlign ksahlin$ cd build
(base) math172l2:build ksahlin$ ls
CMakeCache.txt		CMakeFiles		Makefile		cmake_install.cmake	version.hpp
(base) math172l2:build ksahlin$ make -j8
[  5%] Building CXX object CMakeFiles/salib.dir/src/cmdline.cpp.o
[ 11%] Building CXX object CMakeFiles/salib.dir/src/refs.cpp.o
[ 17%] Building CXX object CMakeFiles/salib.dir/src/pc.cpp.o
[ 23%] Building CXX object CMakeFiles/salib.dir/src/index.cpp.o
[ 29%] Building CXX object CMakeFiles/salib.dir/src/randstrobes.cpp.o
[ 35%] Building CXX object CMakeFiles/salib.dir/src/sam.cpp.o
[ 41%] Building C object CMakeFiles/salib.dir/ext/xxhash.c.o
[ 47%] Building CXX object CMakeFiles/salib.dir/src/aln.cpp.o
/Users/ksahlin/prefix/source/StrobeAlign/src/randstrobes.cpp:44:19: warning: comparison of integers of different signs: 'int' and 'unsigned int' [-Wsign-compare]
    if (q_min_pos == popped_index){ // we popped the previous minimizer, find new brute force
        ~~~~~~~~~ ^  ~~~~~~~~~~~~
/Users/ksahlin/prefix/source/StrobeAlign/src/randstrobes.cpp:76:14: warning: unused variable 'mask' [-Wunused-variable]
    uint64_t mask = (1ULL<<2*k) - 1;
             ^
/Users/ksahlin/prefix/source/StrobeAlign/src/randstrobes.cpp:121:36: warning: comparison of integers of different signs: 'int' and 'std::__1::__deque_base<unsigned int, std::__1::allocator<unsigned int> >::value_type' (aka 'unsigned int') [-Wsign-compare]
                    if (qs_min_pos == qs_pos[t-1]) { // occurs at t:th position in k-mer
                        ~~~~~~~~~~ ^  ~~~~~~~~~~~
/Users/ksahlin/prefix/source/StrobeAlign/src/randstrobes.cpp:140:36: warning: comparison of integers of different signs: 'int' and 'std::__1::__deque_base<unsigned int, std::__1::allocator<unsigned int> >::value_type' (aka 'unsigned int') [-Wsign-compare]
                    if (qs_min_pos == qs_pos[t-1]) { // occurs at t:th position in k-mer
                        ~~~~~~~~~~ ^  ~~~~~~~~~~~
/Users/ksahlin/prefix/source/StrobeAlign/src/randstrobes.cpp:188:303: warning: unused parameter 'seq_start' [-Wunused-parameter]
  ...&string_hashes, const std::vector<unsigned int> &pos_to_seq_choord, uint64_t strobe_hashval, unsigned int &strobe_pos_next, uint64_t &strobe_hashval_next,  unsigned int w_start, unsigned int w_end, uint64_t q, unsigned int seq_start, unsigned int seq_en...
                                                                                                                                                                                                                                    ^
/Users/ksahlin/prefix/source/StrobeAlign/src/randstrobes.cpp:218:60: warning: unused parameter 'n' [-Wunused-parameter]
void seq_to_randstrobes2(ind_mers_vector& flat_vector, int n, int k, int w_min, int w_max, const std::string &seq, int ref_index, int s, int t, uint64_t q, int max_dist)
                                                           ^
/Users/ksahlin/prefix/source/StrobeAlign/src/randstrobes.cpp:221:22: warning: comparison of integers of different signs: 'std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::size_type' (aka 'unsigned long') and 'int'
      [-Wsign-compare]
    if (seq.length() < w_max) {
        ~~~~~~~~~~~~ ^ ~~~~~
/Users/ksahlin/prefix/source/StrobeAlign/src/randstrobes.cpp:316:47: warning: unused parameter 'n' [-Wunused-parameter]
mers_vector_read seq_to_randstrobes2_read(int n, int k, int w_min, int w_max, const std::string& seq, unsigned int ref_index, int s, int t, uint64_t q, int max_dist)
                                              ^
/Users/ksahlin/prefix/source/StrobeAlign/src/randstrobes.cpp:324:21: warning: comparison of integers of different signs: 'unsigned int' and 'int' [-Wsign-compare]
    if (read_length < w_max) {
        ~~~~~~~~~~~ ^ ~~~~~
/Users/ksahlin/prefix/source/StrobeAlign/src/aln.cpp:365:103: warning: comparison of integers of different signs: 'const int' and 'const unsigned int' [-Wsign-compare]
                auto predicate = [c](decltype(open_nams)::value_type const &nam) { return nam.query_e < c; };
                                                                                          ~~~~~~~~~~~ ^ ~
/Users/ksahlin/prefix/source/StrobeAlign/src/aln.cpp:102:9: warning: unused variable 'nr_good_hits' [-Wunused-variable]
    int nr_good_hits = 0, total_hits = 0;
        ^
/Users/ksahlin/prefix/source/StrobeAlign/src/aln.cpp:103:24: warning: unused variable 'no_rep_fw' [-Wunused-variable]
    bool is_rc = true, no_rep_fw = true, no_rep_rc = true;
                       ^
/Users/ksahlin/prefix/source/StrobeAlign/src/aln.cpp:103:42: warning: unused variable 'no_rep_rc' [-Wunused-variable]
    bool is_rc = true, no_rep_fw = true, no_rep_rc = true;
                                         ^
/Users/ksahlin/prefix/source/StrobeAlign/src/aln.cpp:100:322: warning: unused parameter 'ref_seqs' [-Wunused-parameter]
  ...&final_nams, robin_hood::unordered_map< unsigned int, std::vector<hit>> &hits_per_ref, const mers_vector_read &query_mers, const mers_vector &ref_mers, kmer_lookup &mers_index, int k, const std::vector<std::string> &ref_seqs, const std::string &read, un...
                                                                                                                                                                                                                             ^
/Users/ksahlin/prefix/source/StrobeAlign/src/aln.cpp:100:351: warning: unused parameter 'read' [-Wunused-parameter]
  ...unsigned int, std::vector<hit>> &hits_per_ref, const mers_vector_read &query_mers, const mers_vector &ref_mers, kmer_lookup &mers_index, int k, const std::vector<std::string> &ref_seqs, const std::string &read, unsigned int filter_cutoff ){
                                                                                                                                                                                                                  ^
/Users/ksahlin/prefix/source/StrobeAlign/src/aln.cpp:347:27: warning: comparison of integers of different signs: 'int' and 'unsigned int' [-Wsign-compare]
            if (h.query_s > prev_q_start + k) {
                ~~~~~~~~~ ^ ~~~~~~~~~~~~~~~~
/Users/ksahlin/prefix/source/StrobeAlign/src/aln.cpp:617:103: warning: comparison of integers of different signs: 'const int' and 'const unsigned int' [-Wsign-compare]
                auto predicate = [c](decltype(open_nams)::value_type const &nam) { return nam.query_e < c; };
                                                                                          ~~~~~~~~~~~ ^ ~
/Users/ksahlin/prefix/source/StrobeAlign/src/aln.cpp:393:267: warning: unused parameter 'ref_seqs' [-Wunused-parameter]
  ...find_nams(std::vector<nam> &final_nams, robin_hood::unordered_map< unsigned int, std::vector<hit>> &hits_per_ref, mers_vector_read &query_mers, mers_vector &ref_mers, kmer_lookup &mers_index, int k, const std::vector<std::string> &ref_seqs, std::string ...
                                                                                                                                                                                                                                            ^
/Users/ksahlin/prefix/source/StrobeAlign/src/aln.cpp:393:290: warning: unused parameter 'read' [-Wunused-parameter]
  ...&final_nams, robin_hood::unordered_map< unsigned int, std::vector<hit>> &hits_per_ref, mers_vector_read &query_mers, mers_vector &ref_mers, kmer_lookup &mers_index, int k, const std::vector<std::string> &ref_seqs, std::string &read, unsigned int filter_...
                                                                                                                                                                                                                                        ^
/Users/ksahlin/prefix/source/StrobeAlign/src/aln.cpp:599:27: warning: comparison of integers of different signs: 'int' and 'unsigned int' [-Wsign-compare]
            if (h.query_s > prev_q_start + k) {
                ~~~~~~~~~ ^ ~~~~~~~~~~~~~~~~
/Users/ksahlin/prefix/source/StrobeAlign/src/aln.cpp:725:19: warning: comparison of integers of different signs: 'int' and 'std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::size_type' (aka 'unsigned long') [-Wsign-compare]
    for(int i=0; i<One.length(); i++) {
                 ~^~~~~~~~~~~~~
/Users/ksahlin/prefix/source/StrobeAlign/src/aln.cpp:860:11: warning: unused variable 's1' [-Wunused-variable]
    float s1 = n_max.score;
          ^
/Users/ksahlin/prefix/source/StrobeAlign/src/aln.cpp:866:9: warning: unused variable 'mapq' [-Wunused-variable]
    int mapq = 60;
        ^
/Users/ksahlin/prefix/source/StrobeAlign/src/index.cpp:219:23: warning: comparison of integers of different signs: 'int' and 'uint64_t' (aka 'unsigned long long') [-Wsign-compare]
    for (int i = 0; i < sz; ++i) {
                    ~ ^ ~~
/Users/ksahlin/prefix/source/StrobeAlign/src/index.cpp:246:27: warning: comparison of integers of different signs: 'int' and 'unsigned long long' [-Wsign-compare]
        for (int i = 0; i < to_read; ++i) {
                        ~ ^ ~~~~~~~
/Users/ksahlin/prefix/source/StrobeAlign/src/aln.cpp:1126:11: warning: unused variable 's1' [-Wunused-variable]
    float s1 = n_max.score;
          ^
/Users/ksahlin/prefix/source/StrobeAlign/src/aln.cpp:1378:41: warning: comparison of integers of different signs: 'std::__1::vector<std::__1::tuple<int, alignment>, std::__1::allocator<std::__1::tuple<int, alignment> > >::size_type' (aka 'unsigned long') and
      'int' [-Wsign-compare]
        int max_out = alignments.size() < max_secondary ? alignments.size() : max_secondary;
                      ~~~~~~~~~~~~~~~~~ ^ ~~~~~~~~~~~~~
[ 52%] Building CXX object CMakeFiles/salib.dir/ext/ssw_cpp.cpp.o
In file included from /Users/ksahlin/prefix/source/StrobeAlign/src/aln.cpp:1:
In file included from /Users/ksahlin/prefix/source/StrobeAlign/src/aln.hpp:4:
In file included from /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/string:504:
In file included from /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/string_view:175:
In file included from /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/__string:56:
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:715:71: error: invalid operands to binary expression ('const Hit' and 'const Hit')
    bool operator()(const _T1& __x, const _T1& __y) const {return __x < __y;}
                                                                  ~~~ ^ ~~~
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:3920:17: note: in instantiation of member function 'std::__1::__less<Hit, Hit>::operator()' requested here
            if (__comp(*--__last, *__first))
                ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:4101:12: note: in instantiation of function template specialization 'std::__1::__sort<std::__1::__less<Hit, Hit> &, Hit *>' requested here
    _VSTD::__sort<_Comp_ref>(__first, __last, _Comp_ref(__comp));
           ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:4109:12: note: in instantiation of function template specialization 'std::__1::sort<Hit *, std::__1::__less<Hit, Hit> >' requested here
    _VSTD::sort(__first, __last, __less<typename iterator_traits<_RandomAccessIterator>::value_type>());
           ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:4125:12: note: in instantiation of function template specialization 'std::__1::sort<Hit *>' requested here
    _VSTD::sort(__first.base(), __last.base());
           ^
/Users/ksahlin/prefix/source/StrobeAlign/src/aln.cpp:164:10: note: in instantiation of function template specialization 'std::__1::sort<Hit>' requested here
    std::sort(hits_fw.begin(), hits_fw.end());
         ^
/Users/ksahlin/prefix/source/StrobeAlign/src/aln.cpp:19:10: note: candidate function not viable: 'this' argument has type 'const Hit', but method is not marked const
    bool operator< (const Hit& rhs) {
         ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/utility:595:1: note: candidate template ignored: could not match 'pair<type-parameter-0-0, type-parameter-0-1>' against 'const Hit'
operator< (const pair<_T1,_T2>& __x, const pair<_T1,_T2>& __y)
^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/iterator:731:1: note: candidate template ignored: could not match 'reverse_iterator<type-parameter-0-0>' against 'const Hit'
operator<(const reverse_iterator<_Iter1>& __x, const reverse_iterator<_Iter2>& __y)
^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/iterator:1167:1: note: candidate template ignored: could not match 'move_iterator<type-parameter-0-0>' against 'const Hit'
operator<(const move_iterator<_Iter1>& __x, const move_iterator<_Iter2>& __y)
^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/iterator:1539:1: note: candidate template ignored: could not match '__wrap_iter<type-parameter-0-0>' against 'const Hit'
operator<(const __wrap_iter<_Iter1>& __x, const __wrap_iter<_Iter2>& __y) _NOEXCEPT
^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/tuple:1165:1: note: candidate template ignored: could not match 'tuple<type-parameter-0-0...>' against 'const Hit'
operator<(const tuple<_Tp...>& __x, const tuple<_Up...>& __y)
^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/memory:2973:1: note: candidate template ignored: could not match 'unique_ptr<type-parameter-0-0, type-parameter-0-1>' against 'const Hit'
operator< (const unique_ptr<_T1, _D1>& __x, const unique_ptr<_T2, _D2>& __y)
^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/memory:3031:1: note: candidate template ignored: could not match 'unique_ptr<type-parameter-0-0, type-parameter-0-1>' against 'const Hit'
operator<(const unique_ptr<_T1, _D1>& __x, nullptr_t)
^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/memory:3040:1: note: candidate template ignored: could not match 'unique_ptr<type-parameter-0-0, type-parameter-0-1>' against 'const Hit'
operator<(nullptr_t, const unique_ptr<_T1, _D1>& __x)
^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/memory:4811:1: note: candidate template ignored: could not match 'shared_ptr<type-parameter-0-0>' against 'const Hit'
operator<(const shared_ptr<_Tp>& __x, const shared_ptr<_Up>& __y) _NOEXCEPT
^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/memory:4881:1: note: candidate template ignored: could not match 'shared_ptr<type-parameter-0-0>' against 'const Hit'
operator<(const shared_ptr<_Tp>& __x, nullptr_t) _NOEXCEPT
^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/memory:4889:1: note: candidate template ignored: could not match 'shared_ptr<type-parameter-0-0>' against 'const Hit'
operator<(nullptr_t, const shared_ptr<_Tp>& __x) _NOEXCEPT
^
In file included from /Users/ksahlin/prefix/source/StrobeAlign/src/aln.cpp:1:
In file included from /Users/ksahlin/prefix/source/StrobeAlign/src/aln.hpp:4:
In file included from /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/string:504:
In file included from /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/string_view:175:
In file included from /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/__string:56:
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:3706:20: error: no matching function for call to '__sort3'
    unsigned __r = __sort3<_Compare>(__x1, __x2, __x3, __c);
                   ^~~~~~~~~~~~~~~~~
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:3927:20: note: in instantiation of function template specialization 'std::__1::__sort4<std::__1::__less<Hit, Hit> &, Hit *>' requested here
            _VSTD::__sort4<_Compare>(__first, __first+1, __first+2, --__last, __comp);
                   ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:4101:12: note: in instantiation of function template specialization 'std::__1::__sort<std::__1::__less<Hit, Hit> &, Hit *>' requested here
    _VSTD::__sort<_Comp_ref>(__first, __last, _Comp_ref(__comp));
           ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:4109:12: note: in instantiation of function template specialization 'std::__1::sort<Hit *, std::__1::__less<Hit, Hit> >' requested here
    _VSTD::sort(__first, __last, __less<typename iterator_traits<_RandomAccessIterator>::value_type>());
           ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:4125:12: note: in instantiation of function template specialization 'std::__1::sort<Hit *>' requested here
    _VSTD::sort(__first.base(), __last.base());
           ^
/Users/ksahlin/prefix/source/StrobeAlign/src/aln.cpp:164:10: note: in instantiation of function template specialization 'std::__1::sort<Hit>' requested here
    std::sort(hits_fw.begin(), hits_fw.end());
         ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:3666:1: note: candidate template ignored: substitution failure [with _Compare = std::__1::__less<Hit, Hit> &, _ForwardIterator = Hit *]
__sort3(_ForwardIterator __x, _ForwardIterator __y, _ForwardIterator __z, _Compare __c)
^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:3733:20: error: no matching function for call to '__sort4'
    unsigned __r = __sort4<_Compare>(__x1, __x2, __x3, __x4, __c);
                   ^~~~~~~~~~~~~~~~~
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:3930:20: note: in instantiation of function template specialization 'std::__1::__sort5<std::__1::__less<Hit, Hit> &, Hit *>' requested here
            _VSTD::__sort5<_Compare>(__first, __first+1, __first+2, __first+3, --__last, __comp);
                   ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:4101:12: note: in instantiation of function template specialization 'std::__1::__sort<std::__1::__less<Hit, Hit> &, Hit *>' requested here
    _VSTD::__sort<_Comp_ref>(__first, __last, _Comp_ref(__comp));
           ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:4109:12: note: in instantiation of function template specialization 'std::__1::sort<Hit *, std::__1::__less<Hit, Hit> >' requested here
    _VSTD::sort(__first, __last, __less<typename iterator_traits<_RandomAccessIterator>::value_type>());
           ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:4125:12: note: in instantiation of function template specialization 'std::__1::sort<Hit *>' requested here
    _VSTD::sort(__first.base(), __last.base());
           ^
/Users/ksahlin/prefix/source/StrobeAlign/src/aln.cpp:164:10: note: in instantiation of function template specialization 'std::__1::sort<Hit>' requested here
    std::sort(hits_fw.begin(), hits_fw.end());
         ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:3703:1: note: candidate template ignored: substitution failure [with _Compare = std::__1::__less<Hit, Hit> &, _ForwardIterator = Hit *]
__sort4(_ForwardIterator __x1, _ForwardIterator __x2, _ForwardIterator __x3,
^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:3798:5: error: no matching function for call to '__sort3'
    __sort3<_Compare>(__first, __first+1, __j, __comp);
    ^~~~~~~~~~~~~~~~~
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:3935:20: note: in instantiation of function template specialization 'std::__1::__insertion_sort_3<std::__1::__less<Hit, Hit> &, Hit *>' requested here
            _VSTD::__insertion_sort_3<_Compare>(__first, __last, __comp);
                   ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:4101:12: note: in instantiation of function template specialization 'std::__1::__sort<std::__1::__less<Hit, Hit> &, Hit *>' requested here
    _VSTD::__sort<_Comp_ref>(__first, __last, _Comp_ref(__comp));
           ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:4109:12: note: in instantiation of function template specialization 'std::__1::sort<Hit *, std::__1::__less<Hit, Hit> >' requested here
    _VSTD::sort(__first, __last, __less<typename iterator_traits<_RandomAccessIterator>::value_type>());
           ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:4125:12: note: in instantiation of function template specialization 'std::__1::sort<Hit *>' requested here
    _VSTD::sort(__first.base(), __last.base());
           ^
/Users/ksahlin/prefix/source/StrobeAlign/src/aln.cpp:164:10: note: in instantiation of function template specialization 'std::__1::sort<Hit>' requested here
    std::sort(hits_fw.begin(), hits_fw.end());
         ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:3666:1: note: candidate template ignored: substitution failure [with _Compare = std::__1::__less<Hit, Hit> &, _ForwardIterator = Hit *]
__sort3(_ForwardIterator __x, _ForwardIterator __y, _ForwardIterator __z, _Compare __c)
^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:3831:9: error: no matching function for call to '__sort3'
        _VSTD::__sort3<_Compare>(__first, __first+1, --__last, __comp);
        ^~~~~~~~~~~~~~~~~~~~~~~~
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/__config:868:15: note: expanded from macro '_VSTD'
#define _VSTD std::_LIBCPP_ABI_NAMESPACE
              ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:4061:32: note: in instantiation of function template specialization 'std::__1::__insertion_sort_incomplete<std::__1::__less<Hit, Hit> &, Hit *>' requested here
            bool __fs = _VSTD::__insertion_sort_incomplete<_Compare>(__first, __i, __comp);
                               ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:4101:12: note: in instantiation of function template specialization 'std::__1::__sort<std::__1::__less<Hit, Hit> &, Hit *>' requested here
    _VSTD::__sort<_Comp_ref>(__first, __last, _Comp_ref(__comp));
           ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:4109:12: note: in instantiation of function template specialization 'std::__1::sort<Hit *, std::__1::__less<Hit, Hit> >' requested here
    _VSTD::sort(__first, __last, __less<typename iterator_traits<_RandomAccessIterator>::value_type>());
           ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:4125:12: note: in instantiation of function template specialization 'std::__1::sort<Hit *>' requested here
    _VSTD::sort(__first.base(), __last.base());
           ^
/Users/ksahlin/prefix/source/StrobeAlign/src/aln.cpp:164:10: note: in instantiation of function template specialization 'std::__1::sort<Hit>' requested here
    std::sort(hits_fw.begin(), hits_fw.end());
         ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:3666:1: note: candidate template ignored: substitution failure [with _Compare = std::__1::__less<Hit, Hit> &, _ForwardIterator = Hit *]
__sort3(_ForwardIterator __x, _ForwardIterator __y, _ForwardIterator __z, _Compare __c)
^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:3834:9: error: no matching function for call to '__sort4'
        _VSTD::__sort4<_Compare>(__first, __first+1, __first+2, --__last, __comp);
        ^~~~~~~~~~~~~~~~~~~~~~~~
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/__config:868:15: note: expanded from macro '_VSTD'
#define _VSTD std::_LIBCPP_ABI_NAMESPACE
              ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:3703:1: note: candidate template ignored: substitution failure [with _Compare = std::__1::__less<Hit, Hit> &, _ForwardIterator = Hit *]
__sort4(_ForwardIterator __x1, _ForwardIterator __x2, _ForwardIterator __x3,
^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:3837:9: error: no matching function for call to '__sort5'
        _VSTD::__sort5<_Compare>(__first, __first+1, __first+2, __first+3, --__last, __comp);
        ^~~~~~~~~~~~~~~~~~~~~~~~
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/__config:868:15: note: expanded from macro '_VSTD'
#define _VSTD std::_LIBCPP_ABI_NAMESPACE
              ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:3730:1: note: candidate template ignored: substitution failure [with _Compare = std::__1::__less<Hit, Hit> &, _ForwardIterator = Hit *]
__sort5(_ForwardIterator __x1, _ForwardIterator __x2, _ForwardIterator __x3,
^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:3842:5: error: no matching function for call to '__sort3'
    __sort3<_Compare>(__first, __first+1, __j, __comp);
    ^~~~~~~~~~~~~~~~~
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:3666:1: note: candidate template ignored: substitution failure [with _Compare = std::__1::__less<Hit, Hit> &, _ForwardIterator = Hit *]
__sort3(_ForwardIterator __x, _ForwardIterator __y, _ForwardIterator __z, _Compare __c)
^
16 warnings and 8 errors generated.
make[2]: *** [CMakeFiles/salib.dir/src/aln.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
9 warnings generated.
/Users/ksahlin/prefix/source/StrobeAlign/ext/ssw_cpp.cpp:553:19: warning: unused parameter 'score_matrix_size' [-Wunused-parameter]
    const int&    score_matrix_size,
                  ^
1 warning generated.
2 warnings generated.
make[1]: *** [CMakeFiles/salib.dir/all] Error 2
make: *** [all] Error 2

```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -23,11 +23,5 @@ jobs:
           cd build
           make
           make install
-      - if: runner.os == 'Linux'
-        name: Install test dependencies on Linux
-        run: sudo apt-get install samtools
-      - if: runner.os == 'macOS'
-        name: Install test dependencies on macOS
-        run: brew install samtools
       - name: Run
         run: tests/run.sh

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -23,7 +23,11 @@ jobs:
           cd build
           make
           make install
-      - name: Install test dependencies
+      - if: runner.os == 'Linux'
+        name: Install test dependencies on Linux
         run: sudo apt-get install samtools
+      - if: runner.os == 'macOS'
+        name: Install test dependencies on macOS
+        run: brew install samtools
       - name: Run
         run: tests/run.sh

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -11,7 +11,10 @@ jobs:
         run: "! grep -P -R '\\t' src/ tests/*.cpp"
 
   build:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
     steps:
       - uses: actions/checkout@v3
       - name: Build

--- a/src/aln.cpp
+++ b/src/aln.cpp
@@ -16,7 +16,7 @@ struct Hit {
     unsigned int query_e;
     bool is_rc;
 
-    bool operator< (const Hit& rhs) {
+    bool operator< (const Hit& rhs) const {
         return std::tie(count, offset, query_s, query_e, is_rc)
             < std::tie(rhs.count, rhs.offset, rhs.query_s, rhs.query_e, rhs.is_rc);
     }

--- a/src/unused.cpp
+++ b/src/unused.cpp
@@ -1164,7 +1164,7 @@ struct Hit {
     unsigned int query_e;
     bool is_rc;
 
-    bool operator< (const Hit& rhs) {
+    bool operator< (const Hit& rhs) const {
         return std::tie(count, offset, query_s, query_e, is_rc)
             < std::tie(rhs.count, rhs.offset, rhs.query_s, rhs.query_e, rhs.is_rc);
     }

